lib/request.ex

```diff
@@ -1,4 +1,33 @@
 defmodule HTTPServer.Request do
-  defstruct method: :post, path: "", resource: nil, headers: %{}, body: ""
-  @type method :: :get | :post | nil
+  defstruct method: "", path: "", resource: "", headers: %{}, body: ""
+
+  @type t :: %__MODULE__{
+          method: String.t(),
+          path: String.t(),
+          resource: String.t(),
+          headers: %{optional(String.t()) => any},
+          body: String.t()
+        }
```

Comment:
This is pretty cool right here. I like this formatting a little better, not sure what the "best practice" is.
```suggestion
  @type t :: %__MODULE__{
    method: String.t(),
    path: String.t(),
    resource: String.t(),
    headers: %{optional(String.t()) => any},
    body: String.t()
  }
```

Reply:
I like this better too. A lot of the examples I've seen put the [struct literal syntax all on one line](https://hexdocs.pm/elixir/Kernel.html#defstruct/1-types), but I like the returns between and your proposed spacing looks neater. I've been thinking about trying to add [documentation to my modules](https://elixirschool.com/en/lessons/basics/documentation#best-practice-7), but that feels low in priority to getting it working.